### PR TITLE
[azopenai] Add in whisper (audio transcription and translation) support

### PR DESCRIPTION
**Pull request by richardpark-msft**

### CODE REVIEW

File: sdk/ai/azopenai/models.go [hunk 1]
```diff
@@ -618,68 +787,56 @@ type CompletionsUsage struct {
...
+	// An array of details about specific errors that led to this reported error.
+	Details []azureCoreFoundationsError
 
-// ContentFilterResultsViolence - Describes language related to physical actions intended to hurt, injure, damage, or kill
-// someone or something; describes weapons, etc.
-type ContentFilterResultsViolence struct {
-	// REQUIRED; A value indicating whether or not the content has been filtered.
-	Filtered *bool
+	// An object containing more specific information than the current object about the error.
+	Innererror *azureCoreFoundationsErrorInnererror
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenai/models.go:836]: This looks suspicious.

File: sdk/ai/azopenai/custom_client_audio.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+}
+
+// GetAudioTranscription gets transcribed text and associated metadata from provided spoken audio data. Audio will
+// be transcribed in the written language corresponding to the language it was spoken in. Gets transcribed text
+// and associated metadata from provided spoken audio data. Audio will be transcribed in the written language corresponding
+// to the language it was spoken in.
+// If the operation fails it returns an *azcore.ResponseError type.
+//
+// Generated from API version 2023-09-01-preview
+//   - file - The audio data to transcribe. This must be the binary content of a file in one of the supported media formats: flac,
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenai/custom_client_audio.go:41]: Comment doesn't match the param name.

File: sdk/ai/azopenai/custom_client_audio.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	if err != nil {
+		return getAudioTranslationInternalResponse{}, err
+	}
+
+	return getAudioTranslationInternalResponse{AudioTranscription: *at}, nil
+}
+
+// deserializeAudioTranscription handles deserializing the content if it's text/plain
+// or a JSON object.
+func deserializeAudioTranscription(resp *http.Response) (*AudioTranscription, error) {
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenai/custom_client_audio.go:178]: Probably don't need to return `AudioTranscription` by pointer.

File: sdk/ai/azopenai/custom_client_audio.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+// GetAudioTranscription gets transcribed text and associated metadata from provided spoken audio data. Audio will
+// be transcribed in the written language corresponding to the language it was spoken in. Gets transcribed text
+// and associated metadata from provided spoken audio data. Audio will be transcribed in the written language corresponding
+// to the language it was spoken in.
+// If the operation fails it returns an *azcore.ResponseError type.
+//
+// Generated from API version 2023-09-01-preview
+//   - file - The audio data to transcribe. This must be the binary content of a file in one of the supported media formats: flac,
+//     mp3, mp4, mpeg, mpga, m4a, ogg, wav, webm.
+//   - options - getAudioTranscriptionInternalOptions contains the optional parameters for the Client.getAudioTranscriptionInternal
```
[the remark this answers is not in the document]
Comment (richardpark-msft) [reply sdk/ai/azopenai/custom_client_audio.go]: Fixed.

File: sdk/ai/azopenai/custom_client_audio.go (hunk 3 above)
Comment (richardpark-msft) [reply sdk/ai/azopenai/custom_client_audio.go]: Fixed.

File: sdk/ai/azopenai/custom_client_audio.go (hunk 2 above)
Comment (richardpark-msft) [reply sdk/ai/azopenai/custom_client_audio.go]: Fixed.

File: sdk/ai/azopenai/models.go (hunk 1 above)
Comment (richardpark-msft) [reply sdk/ai/azopenai/models.go]: Fixed. That type sneaks in sometimes.

I added some autorest hackery and it looks better now. Also, fixed the InnerErrorInnerError nonsense (the type is actually recursive).

**Comment by richardpark-msft** [comment]
It's beautiful now @jhendrixMSFT.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**